internal/charmstore/store.go

```diff
@@ -71,6 +73,10 @@ type Pool struct {
 // limit specified by config.MaxMgoSessions.
 const reqStoreCacheSize = 50
 
+// maxAsyncGoroutines holds the maximum number
+// of goroutines that will be started by Store.Go.
+const maxAsyncGoroutines = 50
```

Comment:
If this is not part of the configuration, the max-mgo-sessions attribute is not fully functional.